lib/index.js

```diff
@@ -211,27 +214,39 @@ exports.kill = function( pid, signal, next ){
   }
 
   var checkConfident = 0;
+  var checkTimeoutTimer = null;
+  var checkIsTimeout = false;
 
   function checkKilled(finishCallback) {
-    console.log('check kill success', pid, 'confident', checkConfident);
     exports.lookup({ pid: pid }, function(err, list) {
-      console.log('check kill success result:', (list || []).length);
+      if (checkIsTimeout) return;
+
+      console.log(pid, 'check result', list.length);
       if (err) {
-          finishCallback && finishCallback(err);
-        } else if(list.length > 0) {
-          checkKilled(finishCallback);
+        clearTimeout(checkTimeoutTimer);
+        finishCallback && finishCallback(err);
+      } else if(list.length > 0) {
+        checkKilled(finishCallback);
+      } else {
+        checkConfident++;
+        if (checkConfident === 5) {
+          console.log(pid, 'pass check confident');
+          clearTimeout(checkTimeoutTimer);
+          finishCallback && finishCallback();
         } else {
-          checkConfident++;
-          if (checkConfident === 5) {
-            finishCallback && finishCallback();
-          } else {
-            checkKilled(finishCallback);
-          }
+          checkKilled(finishCallback);
         }
+      }
     });
   }
 
   next && checkKilled(next);
+
+  checkTimeoutTimer = next && setTimeout(function() {
```

Comment:
Using setTimeout here seems a little overkill. You could just remember a timestamp when the killing started, then compare the current timestamp against that inside the recursion. That would remove the need for the two variables to keep track of timing out.